apps/web/src/routes/app/editor/agent/+page.server.ts

```diff
@@ -0,0 +1,104 @@
+import { supabase } from '$lib/supabase';
+import { fail } from '@sveltejs/kit';
+import type { Actions, PageServerLoad } from './$types';
+import { agentFormSchema } from '$lib/schema';
+import { superValidate } from 'sveltekit-superforms/server';
+import { pickRandomAvatar } from '$lib/utils';
+
+export const load = (async ({ locals }) => {
+	const session = await locals.getSession();
+	const getCurrentUserAgents = await supabase
+		.from('agents')
+		.select('*')
+		.eq('profile_id', session?.user.id);
+	return {
+		getCurrentUserAgents,
+		agentForm: await superValidate(agentFormSchema)
+	};
+}) satisfies PageServerLoad;
+
+export const actions: Actions = {
+	creatAgents: async ({ request, locals }) => {
+		const session = await locals.getSession();
+
+		const form = await superValidate(request, agentFormSchema);
+
+		if (!form.valid) {
+			return fail(400, { form, message: 'unable to create a new agent' });
+		}
+
+		const randomAvatar = pickRandomAvatar();
+
+		try {
+			const { data, error } = await supabase
+				.from('agents')
+				.insert([
+					{
+						profile_id: session?.user.id,
+						title: form.data.title,
+						description: form.data.description.split(',').map((item: string) => item.trim()),
+						model: form.data.model,
+						role: form.data.role,
+						published: form.data.published,
+						tools: [''],
+						avatar: randomAvatar.avatarUrl,
+						version: '1.0',
+						system_message: ''
+					}
+				])
+				.select();
+
+			if (error) {
+				console.error('Error creating agent:', error);
+				return { error };
+			}
+
+			return fail(200, {
+				message: 'Agent created successfully please reload the page to view your new agent'
+			});
+		} catch (error) {
+			console.error(error);
+			return fail(500, {
+				message: 'Something went wrong , please try again'
+			});
+		}
+	},
+	editAgent: async ({ request, url }) => {
+		const id = url.searchParams.get('id');
+
+		const form = await superValidate(request, agentFormSchema);
+
+		console.log(form, 'from from agent');
+		if (!form.valid) {
+			return fail(400, { form, message: 'Could not edit agent' });
+		}
+
+		try {
```

Comment:
@eksno is try-catch good practice?

Reply:
It is not, only the line at risk of erroring should be in a try catch. Also on another note why are we returning "fail" with the status code "200" indicating success? this looks very weird.